src/Parser.js

```diff
@@ -976,6 +989,7 @@ export default class Parser {
                     firstToken.range(lastToken, firstToken.text));
         } else {
             // Otherwise, just return a nucleus, or nothing for an optional group
+            // TODO: Not yet switching modes for single-symbol group.
```

Comment:
I'm not actually sure how to do it -- it's pretty challenging, because by the time `parseGroup` gets called, `nextToken` *already* has the single-symbol group, which means that it already got expanded in the wrong mode.  In the old way of doing things, we could "unget" the symbol, change modes, and reget it, but that seems challenging/impossible to do if there are side-effects/errors caused by the first execution in the wrong mode.  So I think we'd need to call `parseGroup` before actually reading the first symbol of the group, which might cause a rather big change across all of `Parser.js`.

So in summary, I'm not sure how much work it'll be.  It's surely possible, but hard.  I thought I'd commit what I have so far, and open another issue for fixing this further.  On the plus side, it took me a lot of playing around with tests before I could even get this situation to arise, and in some sense #924 is even more important than this missing case.  So maybe this PR is good to merge for now?  Up to you -- I can look at this more when I have time.